core/src/main/java/org/apache/calcite/sql/dialect/SnowflakeSqlDialect.java

```diff
@@ -42,35 +44,34 @@ public SnowflakeSqlDialect(Context context) {
 
   @Override public void unparseCall(final SqlWriter writer, final SqlCall call, final int leftPrec,
       final int rightPrec) {
+    SqlOperator op;
     switch (call.getKind()) {
     case BIT_AND:
-      SqlCall bitAndCall = SqlLibraryOperators.BITAND_AGG
-          .createCall(SqlParserPos.ZERO, call.getOperandList());
-      super.unparseCall(writer, bitAndCall, leftPrec, rightPrec);
+      op = SqlLibraryOperators.BITAND_AGG;
       break;
     case BIT_OR:
-      SqlCall bitOrCall = SqlLibraryOperators.BITOR_AGG
-          .createCall(SqlParserPos.ZERO, call.getOperandList());
-      super.unparseCall(writer, bitOrCall, leftPrec, rightPrec);
+      op = SqlLibraryOperators.BITOR_AGG;
       break;
     case CHAR_LENGTH:
-      SqlCall lengthCall = SqlLibraryOperators.LENGTH
-          .createCall(SqlParserPos.ZERO, call.getOperandList());
-      super.unparseCall(writer, lengthCall, leftPrec, rightPrec);
+      op = SqlLibraryOperators.LENGTH;
       break;
     case ENDS_WITH:
-      SqlCall endsWithCall = SqlLibraryOperators.ENDSWITH
-          .createCall(SqlParserPos.ZERO, call.getOperandList());
-      super.unparseCall(writer, endsWithCall, leftPrec, rightPrec);
+      op = SqlLibraryOperators.ENDSWITH;
       break;
     case STARTS_WITH:
-      SqlCall startsWithCall = SqlLibraryOperators.STARTSWITH
-          .createCall(SqlParserPos.ZERO, call.getOperandList());
-      super.unparseCall(writer, startsWithCall, leftPrec, rightPrec);
+      op = SqlLibraryOperators.STARTSWITH;
+      break;
+    case MAX:
+      op = SqlStdOperatorTable.MAX;
+      break;
+    case MIN:
+      op = SqlStdOperatorTable.MIN;
       break;
     default:
-      super.unparseCall(writer, call, leftPrec, rightPrec);
+      op = call.getOperator();
```

Comment:
If the call is fine, let's not make a new one.

Reply:
Yeah I didn't like doing that either.. just added a different approach in the second commit